File: Basic_OpenMP_Codes/C_Codes/schedule_clause.c
// OpenMP program to demonstrate the schedule clause
#include <omp.h>
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char* argv[])
{
	int i, id, a[10];
	omp_set_num_threads(3);
	// Arguments for schedule clause can be static, dynamic, guided, auto, runtime
	#pragma omp parallel for private(i, id) shared(a) schedule(static, 2)
	for(i = 0; i < 12; i++)
	{
		id = omp_get_thread_num();
		a[i] = id;
		printf("a[%d] is %d from thread %d \n", i, a[i], id);
	}
}

File: Basic_OpenMP_Codes/FORTRAN_Codes/schedule_clause.f90
! OpenMP program to demonstrate the schedule clause
program schedule_clause

use omp_lib
implicit none

	integer :: i, id
	integer, dimension(:) :: a(10)
	call omp_set_num_threads(3)
	
	! Arguments for schedule clause can be static, dynamic, guided, auto, runtime
	!$omp parallel do private(i, id) shared(a) schedule(static, 2)		
		do i = 1, 10
			id = omp_get_thread_num()
			a(i) = i + 2
			write(*, *) i, a(i), id
		end do
	!$omp end parallel do

end program schedule_clause
